src/test/java/io/vertx/test/core/NetClientTLSConnectTest.java

```diff
@@ -0,0 +1,72 @@
+package io.vertx.test.core;
+
+import java.util.concurrent.CountDownLatch;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.core.net.JksOptions;
+import io.vertx.core.net.NetClient;
+import io.vertx.core.net.NetClientOptions;
+import io.vertx.core.net.NetSocket;
+
+/**
+ * @author <a href="http://oss.lehmann.cx/">Alexander Lehmann</a>
+ *
+ */
+public class NetClientTLSConnectTest extends VertxTestBase {
+
+  private HttpServer server;
+
+  @Before
+  public void startServer() throws InterruptedException {
+    CountDownLatch latch = new CountDownLatch(1);
+
+    HttpServerOptions serverOptions = new HttpServerOptions();
+    serverOptions.setSsl(true);
+    serverOptions.setKeyCertOptions(new JksOptions().setPath("tls/server-keystore-root-ca.jks").setPassword("wibble"));
+    server = vertx.createHttpServer(serverOptions.setPort(4043));
+    server.requestHandler(req -> {
+      req.bodyHandler(buffer -> {
+        req.response().end();
+      });
+    });
+    server.listen(ar -> latch.countDown());
+    latch.await();
+  }
+
+  @After
+  public void stopServer() {
+    server.close();
+  }
+
+  @Test
+  public void testHostnameCheck() {
```

Comment:
rather have this test in NetTest instead of having a new class just for this test